### DISCUSSION TITLE
GA_Pygad

### DISCUSSION THREAD

**Discussion by nstrn-z** [ahmedfgad/GeneticAlgorithmPython#31] [General]
Mr Gad,
I have a question about this error:
ValueError: Unable to coerce to Series, length must be 12: given 10000

When running a genetic algorithm with Pygad, and the Churn_Modelling dataset, I get the following error for this line of code:
output = np.sum(solution * function_inputs, axis = 1)

> function_inputs = dataset.drop(['Exited'], axis = 1)
> desired_output = dataset['Exited']

Help me please? What is the solution to the error?
Thank you for your attention

**Comment by ahmedfgad** [comment]
I think there is a mismatch between the length of these 2 variables:
1. `solution`
2. `function_inputs`

As I do not have the code, I may not be able to locate the error accurately. But for now, could you check the length of `function_inputs ` and `solution ` and make sure the lengths are equal?

**Comment by nstrn-z** [comment]
What does PyGad stand for?
And how is it pronounced?

**Reply by ahmedfgad** [reply]
It is written as PyGAD not PyGad. 

The name is described in the paper about PyGAD: [Gad, Ahmed Fawzy. "PyGAD: An Intuitive Genetic Algorithm Python Library." arXiv preprint arXiv:2106.06158 (2021).](https://arxiv.org/abs/2106.06158) 

Here is the description of the name.
![Capture](https://user-images.githubusercontent.com/16560492/122688517-a8d4c180-d1ea-11eb-851e-9413e021519b.PNG)

Note that it is not intentional to give it my last name. The first suggestion I had for the name is **PyGA** but I already found a library with that name at PyPI. I added the extra character **D** because it was supporting only decimal genetic algorithm. At that time, I figured out that this is my last name :)

For the pronunciation, check these videos:
* 4:19 https://youtu.be/vXHzppq6hzU?t=259
* 1:18 https://youtu.be/QQKYrmF4QUY?t=78

Please let me know if I can help in any way.

**Comment by nstrn-z** [comment]
Thank you for your kind tips and for providing us with complete and useful
information.
I read all the articles and topics related to your genetic algorithm and
neural networks and even used some of them in my dissertation.

**Reply by ahmedfgad** [reply]
Awesome! I hope things are clear in the documentation also.
